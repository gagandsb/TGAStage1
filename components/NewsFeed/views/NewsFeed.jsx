import React, { Component } from 'react';
import styles from '../styles/NewsFeed.style';
import withStyles from '../../hoc/withStyles';
import TimeAgo from 'javascript-time-ago';
import en from 'javascript-time-ago/locale/en';

class NewsFeed extends Component {
  constructor(props) {
    super(props);
    this.state = {
      upVotes: 0,
    };
  }

  renderListItems = (item, index) => {
    const { title, num_comments, objectID, url, author, created_at } = item;
    TimeAgo.addLocale(en);
    const timeAgo = new TimeAgo('en-US');
    const createdAt = timeAgo.format(new Date(created_at));
    return (
      <div className="feeds-wrapper">
        <div className="row feeds-container" key={index}>
          <div className="desktop-comments">
            <div className="comments-column">
              <span>{num_comments}</span>
            </div>
            <div className="upvote-column">
              <span>{this.getLocalStorageValue(objectID)}</span>
              <img
                src="/static/images/arrow-up.gif"
                alt="up arrow"
                className="arrow"
                onClick={() => this.setLocalStorage(objectID)}
              />
            </div>
          </div>
          <div className="col news-title">
            <span className="news-text">{title}</span>
            <a href={url} className="news-subtext-url" target="_blank">
              ({url})
            </a>
            <span className="news-subtext-author">by {author}</span>
            <span className="news-subtext-time">{createdAt}</span>
          </div>
        </div>
        <div className="row mobile-comments" key={index}>
          <div className="comments-column">
            <span>{num_comments}</span>
          </div>
          <div className="upvote-column">
            <span>{this.getLocalStorageValue(objectID)}</span>
            <img
              src="/static/images/arrow-up.gif"
              alt="up arrow"
              className="arrow"
              onClick={() => this.setLocalStorage(objectID)}
            />
          </div>
        </div>
      </div>
    );
  };

  setLocalStorage = (key) => {
    const items = this.getAllNewsData('NewsUpVotes');
    let results = [];
    let item = {};
    let val = 1;
    let index = -1;
    if (items && items.length) {
      results = items;
      const object = results.find((i) => i.name === key);
      index = results.findIndex((i) => i.name === key);
      val = object ? object.value + 1 : 1;
    }
    item.name = key;
    item.value = val;
    if (index !== -1) {
      results[index] = item;
    } else {
      results.push(item);
    }
    localStorage.setItem('NewsUpVotes', JSON.stringify(results));
    const { upVotes } = this.state;
    this.setState({ upVotes: upVotes + 1 });
  };

  getLocalStorageValue = (key) => {
    const results = this.getAllNewsData();
    if (results && results.length) {
      const item = results.find((i) => i.name === key);
      return item && item.name ? item.value : 0;
    }
    return 0;
  };

  getAllNewsData = () => {
    if (this.isClient()) {
      const items = window.localStorage.getItem('NewsUpVotes');
      return JSON.parse(items);
    }
    return [];
  };

  isClient() {
    return typeof window !== 'undefined';
  }

  render() {
    const { feed, className } = this.props;
    return (
      <div className={className}>
        {feed.map((val, index) => this.renderListItems(val, index))}
      </div>
    );
  }
}

export default withStyles(NewsFeed, styles);
export { NewsFeed as NewsFeedVanilla };
