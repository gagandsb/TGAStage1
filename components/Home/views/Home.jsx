import React, { Component } from 'react';
import fetch from 'isomorphic-unfetch';
import NewsFeed from '../../NewsFeed';
import styles from '../styles/Home.style';
import withStyles from '../../hoc/withStyles';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pagesFeed: [],
      page: 0,
    };
  }

  static async getInitialProps() {
    const url = `https://hn.algolia.com/api/v1/search?query=india&tags=story&page=0`;
    const res = await fetch(url);
    const results = await res.json();
    return {
      feed: results.hits,
    };
  }

  fetchNewsFeed = (page = 0) => {
    const url = `https://hn.algolia.com/api/v1/search?query=india&tags=story&page=${page}`;
    fetch(url)
      .then((res) => res.json())
      .then((res) => {
        this.setState({ pagesFeed: res.hits });
      });
  };

  renderNextPage = () => {
    const { page } = this.state;
    const newPage = page + 1;
    this.setState({ page: newPage });
    this.fetchNewsFeed(newPage);
  };

  render() {
    const { pagesFeed } = this.state;
    const { className, feed = [] } = this.props;
    const data = pagesFeed && pagesFeed.length ? pagesFeed : feed;
    return (
      <div className={className}>
        <div className="container-fluid">
          <div className="hacker-news">
            <div className="row header-row">
              <div className="col">
                <a href="#" className="header-text">
                  Hacker News
                </a>
                <a href="#" className="header-subtext">
                  <span>top |</span>
                </a>
                <a href="#" className="header-subtext">
                  <span> new</span>
                </a>
              </div>
            </div>
            <div className="row">
              <NewsFeed feed={data} className="news-wrapper" />
            </div>
            <div className="row footer-row">
              <div className="col">
                <a
                  href="javascript:void(0);"
                  onClick={this.renderNextPage}
                  className="more-link"
                >
                  More
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default withStyles(Home, styles);
export { Home as HomeVanilla };
