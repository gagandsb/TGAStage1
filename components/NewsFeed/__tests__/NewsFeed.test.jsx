import React from 'react';
import { shallow } from 'enzyme';
import { NewsFeedVanilla as NewsFeed } from '../views/NewsFeed';

describe('News Feed component', () => {
  let props;
  let component;
  beforeEach(() => {
    props = {
      className: '',
      feed: [
        {
          "created_at": "2020-04-10T06:17:01.000Z",
          "title": "India heartwarming messages in fight against COVID-19",
          "url": "https://together-india.com/",
          "author": "ankit_mishra",
          "points": 3,
          "story_text": null,
          "comment_text": null,
          "num_comments": 0,
          "story_id": null,
          "story_title": null,
          "story_url": null,
          "parent_id": null,
          "created_at_i": 1586499421,
          "_tags": [
            "story",
            "author_ankit_mishra",
            "story_22830467"
          ],
          "objectID": "22830467",
          "_highlightResult": {
            "title": {
              "value": "<em>India</em> heartwarming messages in fight against COVID-19",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "url": {
              "value": "https://together-<em>india</em>.com/",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "author": {
              "value": "ankit_mishra",
              "matchLevel": "none",
              "matchedWords": []
            }
          }
        },
        {
          "created_at": "2020-04-10T04:44:55.000Z",
          "title": "Dil mil (indian dating app) exposes exact user geolocation",
          "url": "https://medium.com/@kunala/how-i-hacked-dil-mil-indian-dating-app-to-expose-a-users-exact-location-3eb09bfc3fcd",
          "author": "kunala",
          "points": 3,
          "story_text": null,
          "comment_text": null,
          "num_comments": 1,
          "story_id": null,
          "story_title": null,
          "story_url": null,
          "parent_id": null,
          "created_at_i": 1586493895,
          "_tags": [
            "story",
            "author_kunala",
            "story_22830037"
          ],
          "objectID": "22830037",
          "_highlightResult": {
            "title": {
              "value": "Dil mil (<em>india</em>n dating app) exposes exact user geolocation",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "url": {
              "value": "https://medium.com/@kunala/how-i-hacked-dil-mil-<em>india</em>n-dating-app-to-expose-a-users-exact-location-3eb09bfc3fcd",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "author": {
              "value": "kunala",
              "matchLevel": "none",
              "matchedWords": []
            }
          }
        },
        {
          "created_at": "2020-04-09T19:42:39.000Z",
          "title": "How react suspense displays loading indicators",
          "url": "https://twitter.com/dani_akash_/status/1247617443897290752",
          "author": "DaniAkash",
          "points": 1,
          "story_text": null,
          "comment_text": null,
          "num_comments": 0,
          "story_id": null,
          "story_title": null,
          "story_url": null,
          "parent_id": null,
          "created_at_i": 1586461359,
          "_tags": [
            "story",
            "author_DaniAkash",
            "story_22826393"
          ],
          "objectID": "22826393",
          "_highlightResult": {
            "title": {
              "value": "How react suspense displays loading <em>indica</em>tors",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "url": {
              "value": "https://twitter.com/dani_akash_/status/1247617443897290752",
              "matchLevel": "none",
              "matchedWords": []
            },
            "author": {
              "value": "DaniAkash",
              "matchLevel": "none",
              "matchedWords": []
            }
          }
        },
        {
          "created_at": "2020-04-09T18:05:51.000Z",
          "title": "Learning Multi-dimensional indices: The next big thing in OLAP DBs",
          "url": "https://www.kharekartik.dev/2020/04/09/learning-multi-dimensional-indices/",
          "author": "kkcorps",
          "points": 4,
          "story_text": null,
          "comment_text": null,
          "num_comments": 0,
          "story_id": null,
          "story_title": null,
          "story_url": null,
          "parent_id": null,
          "created_at_i": 1586455551,
          "_tags": [
            "story",
            "author_kkcorps",
            "story_22825432"
          ],
          "objectID": "22825432",
          "_highlightResult": {
            "title": {
              "value": "Learning Multi-dimensional <em>indic</em>es: The next big thing in OLAP DBs",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "url": {
              "value": "https://www.kharekartik.dev/2020/04/09/learning-multi-dimensional-<em>indic</em>es/",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "author": {
              "value": "kkcorps",
              "matchLevel": "none",
              "matchedWords": []
            }
          }
        },
        {
          "created_at": "2020-04-09T12:55:21.000Z",
          "title": "Getting to £4k MRR by sharing great user flow examples: PageFlows.com",
          "url": "https://www.indiebeers.co/interviews/page-flows-ramy-khuffash-indiebeers",
          "author": "indiebeers",
          "points": 9,
          "story_text": null,
          "comment_text": null,
          "num_comments": 0,
          "story_id": null,
          "story_title": null,
          "story_url": null,
          "parent_id": null,
          "created_at_i": 1586436921,
          "_tags": [
            "story",
            "author_indiebeers",
            "story_22822045"
          ],
          "objectID": "22822045",
          "_highlightResult": {
            "title": {
              "value": "Getting to £4k MRR by sharing great user flow examples: PageFlows.com",
              "matchLevel": "none",
              "matchedWords": []
            },
            "url": {
              "value": "https://www.<em>indie</em>beers.co/interviews/page-flows-ramy-khuffash-<em>indie</em>beers",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "author": {
              "value": "<em>indie</em>beers",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            }
          }
        },
        {
          "created_at": "2020-04-09T08:33:13.000Z",
          "title": "Mitigating the effects of video call latency through UI",
          "url": null,
          "author": "ohmyblock",
          "points": 1,
          "story_text": "Quick idea: Show a progress bar during the video call indicating how long I have to wait before talking to avoid interrupting the other person? The progress bar speed should change depending on the current latency.",
          "comment_text": null,
          "num_comments": 0,
          "story_id": null,
          "story_title": null,
          "story_url": null,
          "parent_id": null,
          "created_at_i": 1586421193,
          "_tags": [
            "story",
            "author_ohmyblock",
            "story_22820766"
          ],
          "objectID": "22820766",
          "_highlightResult": {
            "title": {
              "value": "Mitigating the effects of video call latency through UI",
              "matchLevel": "none",
              "matchedWords": []
            },
            "author": {
              "value": "ohmyblock",
              "matchLevel": "none",
              "matchedWords": []
            },
            "story_text": {
              "value": "Quick idea: Show a progress bar during the video call <em>indica</em>ting how long I have to wait before talking to avoid interrupting the other person? The progress bar speed should change depending on the current latency.",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            }
          }
        },
        {
          "created_at": "2020-04-09T07:00:33.000Z",
          "title": "Notes on tech from India's largest stockbroker – Bootstrapped and FOSS first",
          "url": "https://zerodha.tech/blog/hello-world/",
          "author": "tango12",
          "points": 2,
          "story_text": null,
          "comment_text": null,
          "num_comments": 0,
          "story_id": null,
          "story_title": null,
          "story_url": null,
          "parent_id": null,
          "created_at_i": 1586415633,
          "_tags": [
            "story",
            "author_tango12",
            "story_22820418"
          ],
          "objectID": "22820418",
          "_highlightResult": {
            "title": {
              "value": "Notes on tech from <em>India</em>'s largest stockbroker – Bootstrapped and FOSS first",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "url": {
              "value": "https://zerodha.tech/blog/hello-world/",
              "matchLevel": "none",
              "matchedWords": []
            },
            "author": {
              "value": "tango12",
              "matchLevel": "none",
              "matchedWords": []
            }
          }
        },
        {
          "created_at": "2020-04-09T06:20:22.000Z",
          "title": "OYO furloughs thousands of employees as revenue drops by over 50%",
          "url": "https://techcrunch.com/2020/04/08/indias-oyo-furloughs-thousands-of-employees-as-revenue-drops-by-over-50/",
          "author": "JumpCrisscross",
          "points": 97,
          "story_text": null,
          "comment_text": null,
          "num_comments": 42,
          "story_id": null,
          "story_title": null,
          "story_url": null,
          "parent_id": null,
          "created_at_i": 1586413222,
          "_tags": [
            "story",
            "author_JumpCrisscross",
            "story_22820263"
          ],
          "objectID": "22820263",
          "_highlightResult": {
            "title": {
              "value": "OYO furloughs thousands of employees as revenue drops by over 50%",
              "matchLevel": "none",
              "matchedWords": []
            },
            "url": {
              "value": "https://techcrunch.com/2020/04/08/<em>india</em>s-oyo-furloughs-thousands-of-employees-as-revenue-drops-by-over-50/",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "author": {
              "value": "JumpCrisscross",
              "matchLevel": "none",
              "matchedWords": []
            }
          }
        },
        {
          "created_at": "2020-04-09T04:49:41.000Z",
          "title": "Actors Who Are So Similar That They Were Indistinguishable from Each Other",
          "url": "https://pictolic.com/en/article/17-actors-who-are-so-similar-that-they-were-indistinguishable-from-each-other",
          "author": "penteston",
          "points": 2,
          "story_text": null,
          "comment_text": null,
          "num_comments": 1,
          "story_id": null,
          "story_title": null,
          "story_url": null,
          "parent_id": null,
          "created_at_i": 1586407781,
          "_tags": [
            "story",
            "author_penteston",
            "story_22819866"
          ],
          "objectID": "22819866",
          "_highlightResult": {
            "title": {
              "value": "Actors Who Are So Similar That They Were <em>Indis</em>tinguishable from Each Other",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "url": {
              "value": "https://pictolic.com/en/article/17-actors-who-are-so-similar-that-they-were-<em>indis</em>tinguishable-from-each-other",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "author": {
              "value": "penteston",
              "matchLevel": "none",
              "matchedWords": []
            }
          }
        },
        {
          "created_at": "2020-04-09T03:39:17.000Z",
          "title": "Land-Grab Universities",
          "url": "https://www.hcn.org/issues/52.4/indigenous-affairs-education-land-grab-universities",
          "author": "axiomdata316",
          "points": 1,
          "story_text": null,
          "comment_text": null,
          "num_comments": 0,
          "story_id": null,
          "story_title": null,
          "story_url": null,
          "parent_id": null,
          "created_at_i": 1586403557,
          "_tags": [
            "story",
            "author_axiomdata316",
            "story_22819554"
          ],
          "objectID": "22819554",
          "_highlightResult": {
            "title": {
              "value": "Land-Grab Universities",
              "matchLevel": "none",
              "matchedWords": []
            },
            "url": {
              "value": "https://www.hcn.org/issues/52.4/<em>indig</em>enous-affairs-education-land-grab-universities",
              "matchLevel": "full",
              "fullyHighlighted": false,
              "matchedWords": [
                "india"
              ]
            },
            "author": {
              "value": "axiomdata316",
              "matchLevel": "none",
              "matchedWords": []
            }
          }
        }
      ]
    };
    component = shallow(<NewsFeed {...props} />);
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
