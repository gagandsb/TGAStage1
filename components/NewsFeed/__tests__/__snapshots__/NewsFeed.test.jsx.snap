// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Feed component should render correctly 1`] = `
<div
  className=""
>
  <div
    className="feeds-wrapper"
  >
    <div
      className="row feeds-container"
      key="0"
    >
      <div
        className="desktop-comments"
      >
        <div
          className="comments-column"
        >
          <span>
            0
          </span>
        </div>
        <div
          className="upvote-column"
        >
          <span>
            0
          </span>
          <img
            alt="up arrow"
            className="arrow"
            onClick={[Function]}
            src="/static/images/arrow-up.gif"
          />
        </div>
      </div>
      <div
        className="col news-title"
      >
        <span
          className="news-text"
        >
          India heartwarming messages in fight against COVID-19
        </span>
        <a
          className="news-subtext-url"
          href="https://together-india.com/"
          target="_blank"
        >
          (
          https://together-india.com/
          )
        </a>
        <span
          className="news-subtext-author"
        >
          by 
          ankit_mishra
        </span>
        <span
          className="news-subtext-time"
        >
          2 hours ago
        </span>
      </div>
    </div>
    <div
      className="row mobile-comments"
      key="0"
    >
      <div
        className="comments-column"
      >
        <span>
          0
        </span>
      </div>
      <div
        className="upvote-column"
      >
        <span>
          0
        </span>
        <img
          alt="up arrow"
          className="arrow"
          onClick={[Function]}
          src="/static/images/arrow-up.gif"
        />
      </div>
    </div>
  </div>
  <div
    className="feeds-wrapper"
  >
    <div
      className="row feeds-container"
      key="1"
    >
      <div
        className="desktop-comments"
      >
        <div
          className="comments-column"
        >
          <span>
            1
          </span>
        </div>
        <div
          className="upvote-column"
        >
          <span>
            0
          </span>
          <img
            alt="up arrow"
            className="arrow"
            onClick={[Function]}
            src="/static/images/arrow-up.gif"
          />
        </div>
      </div>
      <div
        className="col news-title"
      >
        <span
          className="news-text"
        >
          Dil mil (indian dating app) exposes exact user geolocation
        </span>
        <a
          className="news-subtext-url"
          href="https://medium.com/@kunala/how-i-hacked-dil-mil-indian-dating-app-to-expose-a-users-exact-location-3eb09bfc3fcd"
          target="_blank"
        >
          (
          https://medium.com/@kunala/how-i-hacked-dil-mil-indian-dating-app-to-expose-a-users-exact-location-3eb09bfc3fcd
          )
        </a>
        <span
          className="news-subtext-author"
        >
          by 
          kunala
        </span>
        <span
          className="news-subtext-time"
        >
          4 hours ago
        </span>
      </div>
    </div>
    <div
      className="row mobile-comments"
      key="1"
    >
      <div
        className="comments-column"
      >
        <span>
          1
        </span>
      </div>
      <div
        className="upvote-column"
      >
        <span>
          0
        </span>
        <img
          alt="up arrow"
          className="arrow"
          onClick={[Function]}
          src="/static/images/arrow-up.gif"
        />
      </div>
    </div>
  </div>
  <div
    className="feeds-wrapper"
  >
    <div
      className="row feeds-container"
      key="2"
    >
      <div
        className="desktop-comments"
      >
        <div
          className="comments-column"
        >
          <span>
            0
          </span>
        </div>
        <div
          className="upvote-column"
        >
          <span>
            0
          </span>
          <img
            alt="up arrow"
            className="arrow"
            onClick={[Function]}
            src="/static/images/arrow-up.gif"
          />
        </div>
      </div>
      <div
        className="col news-title"
      >
        <span
          className="news-text"
        >
          How react suspense displays loading indicators
        </span>
        <a
          className="news-subtext-url"
          href="https://twitter.com/dani_akash_/status/1247617443897290752"
          target="_blank"
        >
          (
          https://twitter.com/dani_akash_/status/1247617443897290752
          )
        </a>
        <span
          className="news-subtext-author"
        >
          by 
          DaniAkash
        </span>
        <span
          className="news-subtext-time"
        >
          13 hours ago
        </span>
      </div>
    </div>
    <div
      className="row mobile-comments"
      key="2"
    >
      <div
        className="comments-column"
      >
        <span>
          0
        </span>
      </div>
      <div
        className="upvote-column"
      >
        <span>
          0
        </span>
        <img
          alt="up arrow"
          className="arrow"
          onClick={[Function]}
          src="/static/images/arrow-up.gif"
        />
      </div>
    </div>
  </div>
  <div
    className="feeds-wrapper"
  >
    <div
      className="row feeds-container"
      key="3"
    >
      <div
        className="desktop-comments"
      >
        <div
          className="comments-column"
        >
          <span>
            0
          </span>
        </div>
        <div
          className="upvote-column"
        >
          <span>
            0
          </span>
          <img
            alt="up arrow"
            className="arrow"
            onClick={[Function]}
            src="/static/images/arrow-up.gif"
          />
        </div>
      </div>
      <div
        className="col news-title"
      >
        <span
          className="news-text"
        >
          Learning Multi-dimensional indices: The next big thing in OLAP DBs
        </span>
        <a
          className="news-subtext-url"
          href="https://www.kharekartik.dev/2020/04/09/learning-multi-dimensional-indices/"
          target="_blank"
        >
          (
          https://www.kharekartik.dev/2020/04/09/learning-multi-dimensional-indices/
          )
        </a>
        <span
          className="news-subtext-author"
        >
          by 
          kkcorps
        </span>
        <span
          className="news-subtext-time"
        >
          14 hours ago
        </span>
      </div>
    </div>
    <div
      className="row mobile-comments"
      key="3"
    >
      <div
        className="comments-column"
      >
        <span>
          0
        </span>
      </div>
      <div
        className="upvote-column"
      >
        <span>
          0
        </span>
        <img
          alt="up arrow"
          className="arrow"
          onClick={[Function]}
          src="/static/images/arrow-up.gif"
        />
      </div>
    </div>
  </div>
  <div
    className="feeds-wrapper"
  >
    <div
      className="row feeds-container"
      key="4"
    >
      <div
        className="desktop-comments"
      >
        <div
          className="comments-column"
        >
          <span>
            0
          </span>
        </div>
        <div
          className="upvote-column"
        >
          <span>
            0
          </span>
          <img
            alt="up arrow"
            className="arrow"
            onClick={[Function]}
            src="/static/images/arrow-up.gif"
          />
        </div>
      </div>
      <div
        className="col news-title"
      >
        <span
          className="news-text"
        >
          Getting to Â£4k MRR by sharing great user flow examples: PageFlows.com
        </span>
        <a
          className="news-subtext-url"
          href="https://www.indiebeers.co/interviews/page-flows-ramy-khuffash-indiebeers"
          target="_blank"
        >
          (
          https://www.indiebeers.co/interviews/page-flows-ramy-khuffash-indiebeers
          )
        </a>
        <span
          className="news-subtext-author"
        >
          by 
          indiebeers
        </span>
        <span
          className="news-subtext-time"
        >
          20 hours ago
        </span>
      </div>
    </div>
    <div
      className="row mobile-comments"
      key="4"
    >
      <div
        className="comments-column"
      >
        <span>
          0
        </span>
      </div>
      <div
        className="upvote-column"
      >
        <span>
          0
        </span>
        <img
          alt="up arrow"
          className="arrow"
          onClick={[Function]}
          src="/static/images/arrow-up.gif"
        />
      </div>
    </div>
  </div>
  <div
    className="feeds-wrapper"
  >
    <div
      className="row feeds-container"
      key="5"
    >
      <div
        className="desktop-comments"
      >
        <div
          className="comments-column"
        >
          <span>
            0
          </span>
        </div>
        <div
          className="upvote-column"
        >
          <span>
            0
          </span>
          <img
            alt="up arrow"
            className="arrow"
            onClick={[Function]}
            src="/static/images/arrow-up.gif"
          />
        </div>
      </div>
      <div
        className="col news-title"
      >
        <span
          className="news-text"
        >
          Mitigating the effects of video call latency through UI
        </span>
        <a
          className="news-subtext-url"
          href={null}
          target="_blank"
        >
          (
          )
        </a>
        <span
          className="news-subtext-author"
        >
          by 
          ohmyblock
        </span>
        <span
          className="news-subtext-time"
        >
          a day ago
        </span>
      </div>
    </div>
    <div
      className="row mobile-comments"
      key="5"
    >
      <div
        className="comments-column"
      >
        <span>
          0
        </span>
      </div>
      <div
        className="upvote-column"
      >
        <span>
          0
        </span>
        <img
          alt="up arrow"
          className="arrow"
          onClick={[Function]}
          src="/static/images/arrow-up.gif"
        />
      </div>
    </div>
  </div>
  <div
    className="feeds-wrapper"
  >
    <div
      className="row feeds-container"
      key="6"
    >
      <div
        className="desktop-comments"
      >
        <div
          className="comments-column"
        >
          <span>
            0
          </span>
        </div>
        <div
          className="upvote-column"
        >
          <span>
            0
          </span>
          <img
            alt="up arrow"
            className="arrow"
            onClick={[Function]}
            src="/static/images/arrow-up.gif"
          />
        </div>
      </div>
      <div
        className="col news-title"
      >
        <span
          className="news-text"
        >
          Notes on tech from India's largest stockbroker â Bootstrapped and FOSS first
        </span>
        <a
          className="news-subtext-url"
          href="https://zerodha.tech/blog/hello-world/"
          target="_blank"
        >
          (
          https://zerodha.tech/blog/hello-world/
          )
        </a>
        <span
          className="news-subtext-author"
        >
          by 
          tango12
        </span>
        <span
          className="news-subtext-time"
        >
          a day ago
        </span>
      </div>
    </div>
    <div
      className="row mobile-comments"
      key="6"
    >
      <div
        className="comments-column"
      >
        <span>
          0
        </span>
      </div>
      <div
        className="upvote-column"
      >
        <span>
          0
        </span>
        <img
          alt="up arrow"
          className="arrow"
          onClick={[Function]}
          src="/static/images/arrow-up.gif"
        />
      </div>
    </div>
  </div>
  <div
    className="feeds-wrapper"
  >
    <div
      className="row feeds-container"
      key="7"
    >
      <div
        className="desktop-comments"
      >
        <div
          className="comments-column"
        >
          <span>
            42
          </span>
        </div>
        <div
          className="upvote-column"
        >
          <span>
            0
          </span>
          <img
            alt="up arrow"
            className="arrow"
            onClick={[Function]}
            src="/static/images/arrow-up.gif"
          />
        </div>
      </div>
      <div
        className="col news-title"
      >
        <span
          className="news-text"
        >
          OYO furloughs thousands of employees as revenue drops by over 50%
        </span>
        <a
          className="news-subtext-url"
          href="https://techcrunch.com/2020/04/08/indias-oyo-furloughs-thousands-of-employees-as-revenue-drops-by-over-50/"
          target="_blank"
        >
          (
          https://techcrunch.com/2020/04/08/indias-oyo-furloughs-thousands-of-employees-as-revenue-drops-by-over-50/
          )
        </a>
        <span
          className="news-subtext-author"
        >
          by 
          JumpCrisscross
        </span>
        <span
          className="news-subtext-time"
        >
          a day ago
        </span>
      </div>
    </div>
    <div
      className="row mobile-comments"
      key="7"
    >
      <div
        className="comments-column"
      >
        <span>
          42
        </span>
      </div>
      <div
        className="upvote-column"
      >
        <span>
          0
        </span>
        <img
          alt="up arrow"
          className="arrow"
          onClick={[Function]}
          src="/static/images/arrow-up.gif"
        />
      </div>
    </div>
  </div>
  <div
    className="feeds-wrapper"
  >
    <div
      className="row feeds-container"
      key="8"
    >
      <div
        className="desktop-comments"
      >
        <div
          className="comments-column"
        >
          <span>
            1
          </span>
        </div>
        <div
          className="upvote-column"
        >
          <span>
            0
          </span>
          <img
            alt="up arrow"
            className="arrow"
            onClick={[Function]}
            src="/static/images/arrow-up.gif"
          />
        </div>
      </div>
      <div
        className="col news-title"
      >
        <span
          className="news-text"
        >
          Actors Who Are So Similar That They Were Indistinguishable from Each Other
        </span>
        <a
          className="news-subtext-url"
          href="https://pictolic.com/en/article/17-actors-who-are-so-similar-that-they-were-indistinguishable-from-each-other"
          target="_blank"
        >
          (
          https://pictolic.com/en/article/17-actors-who-are-so-similar-that-they-were-indistinguishable-from-each-other
          )
        </a>
        <span
          className="news-subtext-author"
        >
          by 
          penteston
        </span>
        <span
          className="news-subtext-time"
        >
          a day ago
        </span>
      </div>
    </div>
    <div
      className="row mobile-comments"
      key="8"
    >
      <div
        className="comments-column"
      >
        <span>
          1
        </span>
      </div>
      <div
        className="upvote-column"
      >
        <span>
          0
        </span>
        <img
          alt="up arrow"
          className="arrow"
          onClick={[Function]}
          src="/static/images/arrow-up.gif"
        />
      </div>
    </div>
  </div>
  <div
    className="feeds-wrapper"
  >
    <div
      className="row feeds-container"
      key="9"
    >
      <div
        className="desktop-comments"
      >
        <div
          className="comments-column"
        >
          <span>
            0
          </span>
        </div>
        <div
          className="upvote-column"
        >
          <span>
            0
          </span>
          <img
            alt="up arrow"
            className="arrow"
            onClick={[Function]}
            src="/static/images/arrow-up.gif"
          />
        </div>
      </div>
      <div
        className="col news-title"
      >
        <span
          className="news-text"
        >
          Land-Grab Universities
        </span>
        <a
          className="news-subtext-url"
          href="https://www.hcn.org/issues/52.4/indigenous-affairs-education-land-grab-universities"
          target="_blank"
        >
          (
          https://www.hcn.org/issues/52.4/indigenous-affairs-education-land-grab-universities
          )
        </a>
        <span
          className="news-subtext-author"
        >
          by 
          axiomdata316
        </span>
        <span
          className="news-subtext-time"
        >
          a day ago
        </span>
      </div>
    </div>
    <div
      className="row mobile-comments"
      key="9"
    >
      <div
        className="comments-column"
      >
        <span>
          0
        </span>
      </div>
      <div
        className="upvote-column"
      >
        <span>
          0
        </span>
        <img
          alt="up arrow"
          className="arrow"
          onClick={[Function]}
          src="/static/images/arrow-up.gif"
        />
      </div>
    </div>
  </div>
</div>
`;
